text/0000-buildpack-author-subteam.md

```diff
@@ -0,0 +1,125 @@
+
+The Buildpack Author sub-team will follow the same guidelines as other sub-teams:
+
+- include at least one core team member
+- responsible for narrower sets of concerns related to specific aspects of the project
+
+## Language Binding Responsibilties
+These are the responsibilities for maintaining any language binding:
+1. Implementing new APIs in a timely manner
+1. Release process, planning and updates
+1. Triaging issues and supporting users
+1. Documentation
+1. Buildpack Examples using these bindings
+1. Planning and executing improvements
+
+Today, this means taking ownership over the `libcnb` repo doing the above for it.
+
+## Inline Buildpack Responsibilites
+The buildpack sub-team is responsible for sheparding and maintaining [Inline Buildpacks](https://github.com/buildpacks/rfcs/blob/main/text/0048-inline-buildpack.md).
+
+## Collaboration with other teams
+The buildpack sub-team should engage with the learning team in order to:
+1. Improve documentation for buildpack authors
+1. Document buildpack best practices
+
+The buildpack sub-team should provide feedback on RFCs that change the buildpack API with an eye to the potential implications for buildpack authors.
+
+## Future Scope
+The following are examples of additional tools that would fall to the purview of the buildpack team:
+1. Language bindings for other languages
+1. Project Templates for bootstrapping buildpacks
+1. A test harness for integration testing buildpacks
+1. A DSL for writing buildpacks (e.g. [packfile](https://github.com/sclevine/packfile) or similar)
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+- Yet another sub-team
+- May be difficult to sustain maintainers for all these teams
+- May create a silo instead of spreading out the buildpack author concerns
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Do Nothing
+
+If this team isn't made, `libcnb` will continue to live in the implementation team. That sub-team can then staff more focused contributors/maintainers for maintaing that library. Each other "Buildpack Author" focused project/tool will need to find an appropriate home. This will fragment and hurt the focus of having a single team built around it.
+
+## Distribution Team
+
+The distribution team already handles a part of the Buildpack Author experience which is publishing a buildpack and is currently a fairly small team. This would let the two sub-teams combine with expanded scope. This isn't a perfect match though, since the distribution team is well scoped at this point focusing on the registry, interacting with it as users, and operating it.
+
+# Prior Art
+[prior-art]: #prior-art
+
+* [Rust Dev tools team](https://www.rust-lang.org/governance/teams/dev-tools)
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- Who are the maintainers?
+- Would this sub-team own the pack buildpack author tooling (like `pack buildpack new` and inline buildpacks)?
+- What should the team name be?
+
```

Comment:
Would this team also own the `samples` repo, at least with regards to the buildpacks? That is what comes to mind for me. I can see arguments both ways – perhaps the buildpacks sub-team only owns samples that use their lib bindings, or maybe they own all samples entirely (in which case, Windows is a complicated aspect to throw at a new team).

Reply:
@samj1912 and I talked about taking that in at least for the language binding case. I'd love to have input from @jromero, @jkutner, and the rest of the learning subteam that currently owns it today. We also talked about having an `exmaples/` directory in each language binding repo similar to [serenity-rs](https://github.com/serenity-rs/serenity/tree/current/examples).

Reply:
We also discussed this in the learning team sync last week, I think one ways @hone and I were thinking of making this work is that the language bindings could have small composable examples for buildpacks (like how to set env variables, how to create processes) as a reference on how to use these language bindings and is kept up to date with the latest versions of the BP api. Whereas samples could house some higher level e2e buildpack implementations which combine these concepts, for example our sample Java or bundler buildpacks. People could use the examples as a reference and samples as inspiration. We are hoping this could be a combined effort with the learning team.